resolved docker issues + added machines migration

smellslikekeenspirit opened the pull request:
smol PR, gargantuan task really was to get docker rolling locally to be able to run migrations

pjw9963 commented on server/src/server.ts:
```diff
@@ -30,5 +30,5 @@ const server = new ApolloServer({
 const httpServer = createServer(app);
 httpServer.listen({ port: PORT }, (): void =>
-  console.log(`🚀GraphQL-Server is running on http://localhost:3000/graphql`)
+  console.log(`🚀GraphQL-Server is running on http://localhost:z${PORT}/graphql`)
```
Do you need the "z" in-front of the ${PORT} ?

smellslikekeenspirit replied:
oops, typo

pjw9963 commented on server/src/db/migrations/20211102194503_machines_table.ts:
```diff
@@ -0,0 +1,50 @@
+            t.increments('id');
+            t.string('name', 50);
+            t.integer('category');
+            t.foreign('category').references('id').inTable('MachineFamilies');
```
I could be wrong here, but I think since you already added the column `category` you don't need to re-add the column using `t.foreign`. Instead just do `t.integer('category').references('id').inTable('MachineFamilies');`. I tried doing it this way initially too but I had issues with CASCADE not working

smellslikekeenspirit replied:
Yep, that worked. Thanks!!

pjw9963 commented on server/src/db/migrations/20211102194503_machines_table.ts:
```diff
@@ -0,0 +1,50 @@
+            t.string('room', 5);
+            t.date('date_added').defaultTo(knex.fn.now());
+            t.boolean('reserved').defaultTo(false);
+            t.uuid('reserved_by');
```
Would it makes sense to have a separate table for reservations? This implementation would only allow the current reservation to be known, while a reservation table could have past and future reservations for each machine 

for example: 
| reservation_id | machine_id | student_id | starttime | endtime |

pjw9963 reviewed (commented):
Good start. Let me know if my comments make sense

smellslikekeenspirit commented:
@pjw9963 I added the `Reservations` table as well as a better way of handling promises so transaction queries are run properly. Also changed the `id` field in the `Students` table to autoincrement

pjw9963 reviewed (approved):
looks good, i'll start writing my migrations like this too so we don't get that transaction error
